src/wmtk/simplex/link_condition.cpp

```diff
@@ -98,4 +98,93 @@ bool link_condition(const TriMesh& mesh, const Tuple& edge)
     return true;
 }
 
+bool link_condition_closed_tetmesh(const TetMesh& mesh, const Tuple& edge)
+{
+    // for closed mesh, if link(a) \intersect link(b) == link(ab)
+    const Simplex v_a = Simplex::vertex(edge);
+    const Simplex v_b = Simplex::vertex(mesh.switch_tuple(edge, PrimitiveType::Vertex));
+    const Simplex e_ab = Simplex::edge(edge);
+    const SimplexCollection link_a = link(mesh, v_a); // link(a)
+    const SimplexCollection link_b = link(mesh, v_b); // link(b)
+    const SimplexCollection link_ab = link(mesh, e_ab); // link(ab)
+
+    const SimplexCollection link_a_link_b_intersection =
+        SimplexCollection::get_intersection(link_a, link_b);
+
+    return SimplexCollection::are_simplex_collections_equal(link_a_link_b_intersection, link_ab);
+}
+
+bool link_condition(const TetMesh& mesh, const Tuple& edge)
+{
+    // close mesh check
+    if (!link_condition_closed_tetmesh(mesh, edge)) {
+        return false;
+    }
+
+    // add dummy vertex and link it to boundary faces
+    // the check can be convert from link_dummy(a) \intersect link_dummy(b) == link_dummy(ab) to:
+    // if ab is a boundary edge, then if any boundary face incident a shares a common edge with any
+    // boundary face incident b, return false.
+    // if ab is not boundary edge, then if a is not on
+    // boundary or b is not on boundary, return true, otherwise false
+
+    auto get_boundary_faces = [&mesh](const Tuple& _v) {
```

Comment:
I think the code is easier to read if you make that input a `Simplex`. The invocation of that one confused me.